applications/plugins/Communication/components/serverCommunicationVRPN.h

```diff
@@ -0,0 +1,68 @@
+#include "serverCommunication.h"
+
+#include<vrpn_Analog.h>
+#include<vrpn_Button.h>
+#include<vrpn_Tracker.h>
+#include<vrpn_Text.h>
+#include<vrpn_Connection.h>
+#include<vrpn_Configure.h>
+
+#define MAX 1024
+
+namespace sofa
+{
+
+namespace component
+{
+
+namespace communication
+{
+
+class SOFA_COMMUNICATION_API ServerCommunicationVRPN : public ServerCommunication
+{
+public:
+
+    typedef ServerCommunication Inherited;
+    SOFA_CLASS(ServerCommunicationVRPN, Inherited);
+
+    ServerCommunicationVRPN();
+    virtual ~ServerCommunicationVRPN();
+
+    //ArgumentList stringToArgumentList(std::string dataString);
+
+    //////////////////////////////// Factory VRPN type /////////////////////////////////
+    typedef CommunicationDataFactory VRPNDataFactory;
+    VRPNDataFactory* getFactoryInstance();
+    virtual void initTypeFactory() override;
+    /////////////////////////////////////////////////////////////////////////////////
+
+    virtual std::string getArgumentType(std::string value) override;
+    virtual std::string getArgumentValue(std::string value) override;
+
+protected:
+    vrpn_Connection *m_connection;
+    vrpn_Text_Sender *vrpn_text_sender;
```

Comment:
Remove all vrpn_XX_sender, and use VRPN_Base.
std::vector<VRPN_Base*> vrpnSenderList;

Reply:
Actually, for senders, we have different separate methods for the different types which makes it tough for implementation via a common vector. I tried earlier this week but I failed. This works correctly at receiver since we are just appending the values at vector.

Reply:
Actually  it works because we tested it by using only one subscriber. If there is more than one subscriber, vrpnXXXSenders will be overwrited.

the only thing we do on senders is mainloop.